PhysicsTools/PatAlgos/python/jetCollectionTools.py

```diff
@@ -0,0 +1,621 @@
+        size     = sizeKey,
+      )
+    )
+
+    jetMatch = jetRegex.match(jet.lower())
+    if not jetMatch:
+      raise RuntimeError('Invalid jet collection: %s' % jet)
+    self.jetAlgo     = jetMatch.group(algoKey)
+    self.jetSize     = jetMatch.group(sizeKey)
+    self.jetSizeNr = float(self.jetSize) / 10.
+
+#============================================
+#
+# GenJetAdder
+#
+#============================================
+class GenJetAdder(object):
+  def __init__(self):
+    self.prerequisites = []
+    self.main = []
+    self.gpLabel = "prunedGenParticles"
+
+  def getSequence(self, proc):
+    tasks = self.prerequisites + self.main
+
+    resultSequence = cms.Sequence()
+    for idx, task in enumerate(tasks):
+      if idx == 0:
+        resultSequence = cms.Sequence(getattr(proc, task))
+      else:
+        resultSequence.insert(idx, getattr(proc, task))
+    return resultSequence
+  
+  def addGenJetCollection(self,
+      proc,
+      jet,
+      name,
+      doc,
+      inputCollection    = "",
+      genName            = "",
+      minPt              = 5.,
+    ):
+    print("custom_jme_cff::GenJetAdder::addGenJetCollection: Adding Gen Jet Collection: {}".format(jet))
+    currentTasks = []
+    #
+    # Decide which jet collection we're dealing with
+    #
+    jetLower = jet.lower()
+    jetUpper = jet.upper()
+    tagName  = jetUpper
+    genJetInfo = GenJetInfo(jet,inputCollection)
+
+    #
+    # Skip AK4GenJets and AK8GenJets.
+    # They're already available in the default NanoAOD
+    #
+    if genJetInfo.jetSize  == "4" and genJetInfo.jetAlgo == "ak":
+      pass
+    elif genJetInfo.jetSize  == "8" and genJetInfo.jetAlgo == "ak":
+      pass
+    #=======================================================
+    #
+    # If jet collection in MiniAOD is  not 
+    # specified, build the jet collection.
+    #
+    #========================================================
+    if not inputCollection:
+      print("custom_jme_cff::GenJetAdder::addGenJetCollection: inputCollection in NanoAOD not specified. Building genjet collection now")
+      #
+      # Setup Gen Particles
+      #
+      packedGenPartNoNu = "packedGenParticlesForJetsNoNu"
+      if packedGenPartNoNu not in self.prerequisites:
+        setattr(proc, packedGenPartNoNu, cms.EDFilter("CandPtrSelector",
+            src = cms.InputTag("packedGenParticles"),
+            cut = cms.string("abs(pdgId) != 12 && abs(pdgId) != 14 && abs(pdgId) != 16"),
+          )
+        )
+        self.prerequisites.append(packedGenPartNoNu)    
+      #
+      # Create the GenJet collection
+      #
+      genJetsCollection = "{}{}{}".format(genJetInfo.jetAlgo.upper(), genJetInfo.jetSize, 'GenJetsNoNu')
+      setattr(proc, genJetsCollection,
+        cms.EDProducer("FastjetJetProducer",
+          GenJetParameters.clone(
+            src          = packedGenPartNoNu,
+            doAreaFastjet = cms.bool(True),
+          ),
+          AnomalousCellParameters,
```

Comment:
how much of this is the boilerplate from ak4GenJets_cfi or similar "standard" jet definition?
It seems more practical to do define clone from there and replace only what's necessary.